app/src/main/java/woowacourse/shopping/ui/detail/ProductDetailActivity.kt

```diff
@@ -0,0 +1,94 @@
+package woowacourse.shopping.ui.detail
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.ViewModelProvider
+import com.bumptech.glide.Glide
+import woowacourse.shopping.R
+import woowacourse.shopping.databinding.ActivityProductDetailBinding
+import woowacourse.shopping.model.data.CartsImpl
+import woowacourse.shopping.model.data.ProductsImpl
+
+class ProductDetailActivity : AppCompatActivity(), CartButtonClickListener {
+    private lateinit var binding: ActivityProductDetailBinding
+    private var toast: Toast? = null
+    private val viewModel by lazy {
+        ViewModelProvider(this, ProductDetailViewModelFactory(ProductsImpl, CartsImpl))
+            .get(ProductDetailViewModel::class.java)
+    }
+    private val productId by lazy { productId() }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_product_detail)
+
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_product_detail)
```

Comment:
DataBindingUtil과 ~~Binding을 활용해서 inflate하는 것은 각각 어떤 차이점이 있을까요?

Reply:
```
binding = ActivityProductDetailBinding.inflate(layoutInflater)
setContentView(binding.root)
```
과 같이 사용할 수도 있습니다! 제가 기존에 사용한 DataBindingUtil을 사용하는 방법은 layout의 id를 모르는 경우에만 사용하라고 공식문서의 DataBindingUtil 부분에도 설명이 되어 있네요.. https://developer.android.com/reference/android/databinding/DataBindingUtil

```
T inflate ([LayoutInflater](http://developer.android.com/reference/android/view/LayoutInflater.html) inflater, 
                int layoutId, 
                [ViewGroup](http://developer.android.com/reference/android/view/ViewGroup.html) parent, 
                boolean attachToParent, 
                [DataBindingComponent]

```
```
Inflates a binding layout and returns the newly-created binding for that layout.

Use this version only if layoutId is unknown in advance. Otherwise, use the generated Binding's inflate method to ensure type-safe inflation.
```

사실 '레이아웃id를 모를 때' 가 무엇을 의미하는지 정확히 모르겠습니다. 어플리케이션의 실행 도중에 어떤 레이아웃이 선택될지가 동적으로 결정될 경우를 말하는 것일까요?

Reply:
> 어플리케이션의 실행 도중에 어떤 레이아웃이 선택될지가 동적으로 결정될 경우를 말하는 것일까요?

저도 이 경우가 생각이 나네요!
액티비티 처럼 너무나 명확하게 레이아웃 id에 대해 알고 있다면 DataBindingUtil를 사용할 필요성이 느껴지지 않아서요 :)